src/modules/delete.js

```diff
@@ -4,14 +4,12 @@ const saveData = () => {
 };
 
 export default function clearCheckedItems() {
-  const listItems = document.getElementById('list-container').children;
+  const listContainer = document.getElementById('list-container');
+  const listItems = Array.from(listContainer.children);
+  const uncheckedItems = listItems.filter((item) => !item.classList.contains('checked'));
 
-  for (let i = 0; i < listItems.length; i += 1) {
-    if (listItems[i].classList.contains('checked')) {
-      listItems[i].remove();
-      i -= 1;
-    }
-  }
+  listContainer.innerHTML = '';
+  uncheckedItems.forEach((item) => listContainer.appendChild(item));
 
   saveData();
-}
```

Comment:
### After checking the second item (Look at the localStorage)

![image](https://github.com/AleWaweru/TODO-LIST/assets/93039359/23cc9a64-2b4a-4d30-b534-894e1b5f63c8)

### After removing the second item (Look at the localStorage)

![image](https://github.com/AleWaweru/TODO-LIST/assets/93039359/9abb53f8-c1c9-4d93-b489-c5c9474095ad)

- [ ] As you can see that even after removing one todo it still exists in your localStorage 😨 Kindly filter all your completed todos from your local storage when you click on the clear all completed button as it was requested by the previous reviewer 👍

Reply:
done